packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -254,6 +254,27 @@ bool _writeFlutterPluginsList(
   return pluginsChanged;
 }
 
+/// Checks if the .flutter-plugins-dependencies file has any plugin
+/// dev dependencies with platform-specific implementations.
+bool flutterPluginsListHasDevDependencies(File pluginsFile) {
```

Comment:
Since you're parsing `.flutter-plugins-dependencies` anyways, did you consider writing the `FLUTTER_DEV_DEPENDENCIES_ENABLED` to this file rather than the generated xcconfig?

Reply:
Great question. `.flutter-plugins-dependencies` is arguably the correct place to put plugin information that's used by the native build system. 

In my opinion, `FLUTTER_DEV_DEPENDENCIES_ENABLED` is a gross hack that's needed only on iOS and macOS. I'd prefer to limit this hack to just iOS and macOS if possible by putting it in the `ios/` and `macos` platform directories. If we expose this information in `.flutter-plugins-dependencies`, it'd be easier for other platforms to rely on this information.

That said, putting this in `.flutter-plugins-dependencies` would allow us to remove 1 input from the target. I'm not too concerned by that though.

What are your thoughts?

Reply:
@camsim99 Doesn't android have this same problem where the user can change the configuration via Android Studio and therefore it may/may not have dev dependencies excluded correctly?

Reply:
See response here: https://github.com/flutter/flutter/pull/165916#issuecomment-2797831163